extensions/amp-fx-collection/0.1/test/test-amp-fx-collection.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpFxCollection} from '../amp-fx-collection';
+import {isExperimentOn, toggleExperiment} from '../../../../src/experiments';
+
+describes.realWin('Creates the relevant fx presets correctly', {
+  amp: {
+    ampdoc: 'single',
+    extensions: ['amp-fx-collection'],
+  },
+}, env => {
+  let win;
+  let ampdoc;
+
+  beforeEach(() => {
+    win = env.win;
+    ampdoc = env.ampdoc;
+    toggleExperiment(win, 'amp-fx-fade-in', true, false);
+    toggleExperiment(win, 'amp-fx-fade-in-scroll', true, false);
+  });
+
+  function createAmpFx(fxType, opt_attrs) {
+    const element = win.document.createElement('div');
```

Comment:
You don't need to roll this yourself, there's `createElementWithAttributes` in `src/dom`